### PR TITLE
feat(node): extract `NodePackage` from `NodeProject`

### PR DESCRIPTION
**Pull request by eladb**
In an effort to componentize the monolithic `NodeProject` so that the various aspects
of it can be reused by other projects through composition, this commit extracts the
management of `package.json` into `NodePackage`.

BREAKING CHANGE: `Version.resolveVersion(outdir)` is now `Version.version`.
* `NodeProjectCommonOptions` is now `NodePackageOptions`.

### CODE REVIEW

File: test/npm-package.test.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { NpmPackage } from '../src';
+import { synthSnapshot, TestProject } from './util';
+
+test('minimal', () => {
+  // GIVEN
+  const project = new TestProject();
+
+  // WHEN
```
[the remark this answers is not in the document]
Comment (eladb) [reply test/npm-package.test.ts:8]: Subsequent PR

**Comment by pgollucci** [comment]
obsoletes #369

**Comment by pgollucci** [comment]
I think this unblocks #367

**Comment by eladb** [comment]
> I think this unblocks #367

I am not sure, we still need to add `project.name`, but I'll pick this up next.

**Pull request merged**